src/Microsoft.AspNet.OData.Shared/IDeltaSetItem.cs

```diff
@@ -0,0 +1,38 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="IDeltaSetItem.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Basic interface for representing a delta item like delta, deleted entity, etc.
+    /// </summary>
+    public interface IDeltaSetItem
+    {
+        /// <summary>
+        /// Gets the kind of object within the delta payload.
+        /// </summary>
+        EdmDeltaEntityKind DeltaKind { get; }
+
+        /// <summary>
+        /// Gets or sets the annotation container to hold transient instance annotations.
+        /// </summary>
+        IODataInstanceAnnotationContainer TransientInstanceAnnotationContainer { get; set; }
+
+        /// <summary>
+        /// Gets or sets the container to hold ODataId
+        /// </summary>
+        IODataIdContainer ODataIdContainer { get; set; }
+
+        /// <summary>
+        /// Gets or sets the OData path for the item.
+        /// </summary>
+        ODataPath ODataPath { get; set; }
```

Comment:
>ODataPath [](http://example.com/codeflow?start=18&length=9)

what's the usage of "ODataPath"? shall we need to make it simple as "string" ?

Reply:
The `ODatPath` is used by customers to determine the handler to return. So we pass the `ODataPath` to the customer's handler methods then they return the right `ApiHandler` to be used in patching resources. Passing a string will mean that at some point we'll have to parse the string to create an `ODataPath` or something like that.

Reply:
why is required to determine "ApiHandler"?

Can we use other things to replace it? 

I don't want to let customer to build an ODataPath here?

Reply:
Customers are not creating the APIHandler. We just pass it to them then we also have some extension methods that they can use to extract some methods from the ODataPath,